.circleci/config.yml

```diff
@@ -1,23 +1,458 @@
+          no_output_timeout: 30m
+          command: bundle exec fastlane android build || exit 1
+
+  build-ios:
+    description: "Build the iOS app"
+    steps:
+      - checkout:
+          path: ~/mattermost-mobile
+      - npm-dependencies
+      - assets
+      - ios-gem-dependencies
+      - run:
+          working_directory: fastlane
+          name: Run fastlane to build iOS
+          no_output_timeout: 30m
+          command: bundle exec fastlane ios build || exit 1
+
+  deploy-android:
+    description: "Deploy apk to Google Play"
+    parameters:
+      apk_path:
+        type: string
+    steps:
+      - attach_workspace:
+          at: ~/mattermost-mobile
+      - deploy:
+          name: "Deploy apk to Google Play"
+          working_directory: fastlane
+          command: bundle exec fastlane android deploy apk:../<<parameters.apk_path>>
+
+  deploy-ios:
+    description: "Deploy ipa to TestFlight"
+    parameters:
+      ipa_path:
+        type: string
+    steps:
+      - attach_workspace:
+          at: ~/mattermost-mobile
+      - deploy:
+          name: "Deploy ipa to TestFlight"
+          working_directory: fastlane
+          command: bundle exec fastlane ios deploy ipa:../<<parameters.ipa_path>>
+
+  persist:
+    description: "Persist mattermost-mobile directory"
+    steps:
+      - persist_to_workspace:
+          root: ./
+          paths:
+            - ./
+
+  save:
+    description: "Save binaries artifacts"
+    parameters:
+      filename:
+        type: string
+    steps:
+      - store_artifacts:
+          path: ~/mattermost-mobile/<<parameters.filename>>
 
 jobs:
   test:
     working_directory: ~/mattermost-mobile
     docker:
       - image: circleci/node:10
     steps:
-      - checkout
-      - run: |
-          echo assets/base/config.json
-          cat assets/base/config.json
-          # Avoid installing pods
-          touch .podinstall
-          # Run tests
-          make test || exit 1
+      - checkout:
+          path: ~/mattermost-mobile
+      - npm-dependencies
+      - assets
+      - run:
+          name: Check styles
+          command: npm run check
+      - run:
+          name: Running Tests
+          command: npm test
+
+  build-android-beta:
+    executor: android
+    environment:
+      APP_NAME: "Mattermost Beta"
+      REPLACE_ASSETS: false
+      MAIN_APP_IDENTIFIER: "com.mattermost.rnbeta"
+      SENTRY_ENABLED: true
+      SEGMENT_API_KEY: ${SEGMENT_API_KEY_BETA}
+      TELEMETRY_ENABLED: true
+      AWS_FOLDER_NAME: 'mattermost-mobile-beta'
+      MATTERMOST_WEBHOOK_URL: ${MATTERMOST_WEBHOOK_BUILD}
+    steps:
+      - build-android
+      - persist
+      - save:
+          filename: "Mattermost_Beta.apk"
+
+  build-android-release:
+    executor: android
+    environment:
+      APP_NAME: "Mattermost"
+      REPLACE_ASSETS: true
+      MAIN_APP_IDENTIFIER: "com.mattermost.rn"
+      SENTRY_ENABLED: true
+      SEGMENT_API_KEY: ${SEGMENT_API_KEY_PROD}
```

Comment:
if you are using different context, you wouldn't need to re-define, just name it apropiately in the context.